src/commands/branch.rs

```diff
@@ -293,15 +294,39 @@ fn cmd_branch_list(
 
     let mut formatter = ui.stdout_formatter();
     let formatter = formatter.as_mut();
-    for (name, branch_target) in repo.view().branches() {
+
+    let mut all_branches = repo.view().branches().clone();
+    for (branch_name, git_tracking_target) in git_tracking_branches(repo.view()) {
+        let branch_target = all_branches
+            .entry(branch_name.to_owned())
+            .or_insert(BranchTarget {
+                local_target: None,
+                remote_targets: BTreeMap::default(),
+            });
+        // TODO: This is not currently tested
+        // TODO: Use `BTreeMap::try_insert` once that's stable. That will allow removing
+        // the unwrap()
+        branch_target
+            .remote_targets
+            .entry("git".to_string())
+            .and_modify(|_| {
+                writeln!(
+                    formatter.labeled("warning"),
+                    "WARNING: Branch {branch_name} has a remote-tracking branch for a remote \
+                     named `git`. Local-git tracking branches for it will not be shown.\nIt is \
+                     recommended to rename that remote, as jj normally reserves the `@git` suffix \
+                     to denote local-git tracking branches."
+                )
+                .unwrap();
+            })
+            .or_insert_with(|| git_tracking_target.clone());
+    }
```

Comment:
I think I would prefer to instead ignore a remote called `git` when importing refs. We would ideally print a warning about it too. Or maybe we even make it an error for now, since that's probably easier to implement. What do you think?

Reply:
Sounds like a good idea to me. I put a TODO for now, should also probably file a bug.